src/core/RecyclerListView.tsx

```diff
@@ -684,27 +711,37 @@ export default class RecyclerListView<P extends RecyclerListViewProps, S extends
         if (this.props.onScroll) {
             this.props.onScroll(rawEvent, offsetX, offsetY);
         }
-        this._processOnEndReached();
+        this._processOnEdgeReached();
     }
 
-    private _processOnEndReached(): void {
-        if (this.props.onEndReached && this._virtualRenderer) {
-            const layout = this._virtualRenderer.getLayoutDimension();
+    private _processOnEdgeReached = (): void => {
+        if (!this._getOnEdgeReachedCalled() && this._virtualRenderer && (this.props.onEndReached || this.props.onStartReached)) {
+            const virtualLayout = this._virtualRenderer.getLayoutDimension();
             const viewabilityTracker = this._virtualRenderer.getViewabilityTracker();
             if (viewabilityTracker) {
-                const windowBound = this.props.isHorizontal ? layout.width - this._layout.width : layout.height - this._layout.height;
-                const lastOffset = viewabilityTracker ? viewabilityTracker.getLastOffset() : 0;
-                if (windowBound - lastOffset <= Default.value<number>(this.props.onEndReachedThreshold, 0)) {
-                    if (this.props.onEndReached && !this._onEndReachedCalled) {
-                        this._onEndReachedCalled = true;
-                        this.props.onEndReached();
-                    }
-                } else {
-                    this._onEndReachedCalled = false;
+                const windowBound = this.props.isHorizontal ? virtualLayout.width - this._layout.width : virtualLayout.height - this._layout.height;
+                const lastOffset = viewabilityTracker.getLastOffset();
+                const isWithinEndThreshold = windowBound - lastOffset <= Default.value<number>(this.props.onEndReachedThreshold, 0);
+                const isWithinStartThreshold = lastOffset <= Default.value<number>(this.props.onStartReachedThreshold, 0);
+                if (this.props.onEndReached && isWithinEndThreshold) {
+                    this._onEndReachedCalled = true;
+                    this.props.onEndReached();
+                } else if (this.props.onStartReached && isWithinStartThreshold) {
+                    this._onStartReachedCalled = true;
+                    this.props.onStartReached();
                 }
             }
         }
     }
+
+    private _getOnEdgeReachedCalled(): boolean {
+        return this._onStartReachedCalled || this._onEndReachedCalled;
+    }
+
+    private _setOnEdgeReachedCalled(onEdgeReachedCalled: boolean): void {
+        this._onStartReachedCalled = onEdgeReachedCalled;
+        this._onEndReachedCalled = onEdgeReachedCalled;
+    }
```

Comment:
doesn't it matter which edge we've reached?
We can simultaneously set that we have reached no edge - the `false` case.
But if we're nearing the start edge we can't be near the end edge, why would we set `true` to both?

I think this is only used as a reset, so we could name it `resetOnEdgeReached` and set the values to `false`

Reply:
Good suggestion, it is indeed only used as a reset.